modules/loans/src/lib.rs

```diff
@@ -89,6 +89,12 @@ decl_module! {
 }
 
 impl<T: Trait> Module<T> {
+	pub fn debits_iterator_with_collateral_prefix(
+		currency_id: CurrencyIdOf<T>,
+	) -> PrefixIterator<(T::DebitBalance, Option<(CurrencyIdOf<T>, T::AccountId)>)> {
```

Comment:
I want to modify `PrefixIterator` in Substrate to make it return Keys as well but I guess for now we will just copy and modify the `PrefixIterator` from Substrate and later switch to whatever Substrate provides

https://github.com/paritytech/substrate/issues/4959